lib/puppet/module/task.rb

```diff
@@ -61,6 +62,33 @@ def self.is_tasks_filename?(path)
       return true
     end
 
+    def self.find_files(name, metadata)
+      if metadata.key?('files')
+        file_list = metadata['files'].map do |file|
+          splits = file.split("/")
+          pup_module = Puppet::Module.find(splits[0])
+          if pup_module.nil?
+            Puppet.warning(_("Could not find module %{module}" % { module: splits[0] }))
```

Comment:
Should this include the current environment in the warning? Might make it easier to debug.

Reply:
Yes, it will

Reply:
I think I addressed this as well.